gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/task/SignModule.kt

```diff
@@ -228,28 +338,37 @@ open class SignModule @Inject constructor(_providers: ProviderFactory, _objects:
     @Suppress("MemberVisibilityCanBePrivate")
     @Throws(IOException::class)
     protected fun signModule(
-        keyStoreFile: File,
+        keyStoreFile: File?,
+        pkcs11CfgFile: File?,
         keystorePassword: String,
         cert: File,
         certPassword: String,
         certAlias: String,
         unsignedModule: File,
-        outFile: File
+        outFile: File,
     ) {
         logger.debug(
-            "Signing module with keystoreFile: ${keyStoreFile.absolutePath}, " +
-                "keystorePassword: ${"*".repeat(keystorePassword.length)}, " +
+            "Signing module with keystoreFile: ${keyStoreFile?.absolutePath}, " +
+                "pkcs11CfgFile: ${pkcs11CfgFile?.absolutePath}, " +
+                "keystorePassword: ${"*".repeat(20)}, " +
                 "cert: ${cert.absolutePath}, " +
-                "certPw: ${"*".repeat(certPassword.length)}, " +
+                "certPw: ${"*".repeat(20)}, " +
                 "certAlias: $certAlias"
         )
 
         val keyStore: KeyStore = getKeyStore()
-        keyStore.load(keyStoreFile.inputStream(), keystorePassword.toCharArray())
+        loadKeyStore(keyStore, keystorePassword, keyStoreFile)
 
-        val privateKey: RSAPrivateKey = keyStore.getKey(certAlias, certPassword.toCharArray()) as RSAPrivateKey
+        val privateKey: PrivateKey = keyStore.getKey(certAlias, certPassword.toCharArray()) as PrivateKey
 
         ModuleSigner(privateKey, cert.inputStream())
             .signModule(PrintStream(OutputStream.nullOutputStream()), unsignedModule, outFile)
     }
+
+    private fun loadKeyStore(ks: KeyStore, ksPwd: String, ksFile: File?) {
```

Comment:
See comment above in `getKeyStore` about refactoring this away in favor of logic already in `module-signer`. Something for a future ticket.